### ISSUE TITLE
Stroke-width and BoundingBox

### ISSUE THREAD

**Issue by LucaPassarella** [ElinamLLC/SharpVectors#157]
Hi,
I've a simple text with the stroke-width parameter set at "0.001px", when I try to render this into the WpfTestSvgSample I have a bounding box bigger that my text. If I set the parameter at "0.002px" or bigger the issue disappear.
I try to find the error in the code but I still don't know where it could be.

I try to solve it by adding a simple line into the GetStrokeWidth() function
`strokeWidthLength.Value = Math.Max(strokeWidthLength.Value, 0.002);`
but I know it's an orrible solution.

Have you any solution for this?

Thanks

<img width="659" alt="M1OUHAVlbQ" src="https://user-images.githubusercontent.com/17331427/98379346-7a351c80-2047-11eb-89e8-7015881efda0.png">

[text_test.zip](https://github.com/ElinamLLC/SharpVectors/files/5501819/text_test.zip)

**Comment by paulushub** [comment]
@LucaPassarella It could a simply tricky situation! Looking at the output of both the results for `0.001px` and `0.002px`, the XAML only show difference in the pen width.
```csharp
<Pen Brush="#FFAAAAAA" Thickness="0.001" StartLineCap="Flat" EndLineCap="Flat" LineJoin="Round" />

<Pen Brush="#FFAAAAAA" Thickness="0.002" StartLineCap="Flat" EndLineCap="Flat" LineJoin="Round" />
```
My guess is that the problem is coming from the translate transformation applied to the text, which has not effect for the `0.001px` case,.
```csharp
<DrawingGroup>
  <DrawingGroup x:Name="_SvgDrawingLayer" svg:SvgLink.Key="_SvgDrawingLayer">
    <DrawingGroup.Transform>
      <TranslateTransform X="-91.2802505493164" Y="-78.5458755493164" />
    </DrawingGroup.Transform>
    <DrawingGroup x:Name="id000001">
      <DrawingGroup x:Name="text2985" svg:SvgObject.Type="Text">
        <GeometryDrawing Brush="#FF000000">
          <GeometryDrawing.Pen>
            <Pen Brush="#FFAAAAAA" Thickness="0.002" StartLineCap="Flat" EndLineCap="Flat" LineJoin="Round" />
          </GeometryDrawing.Pen>
          <GeometryDrawing.Geometry>
            <GeometryGroup FillRule="Nonzero">
                        // Text Path Geometry....
            </GeometryGroup>
          </GeometryDrawing.Geometry>
        </GeometryDrawing>
      </DrawingGroup>
    </DrawingGroup>
  </DrawingGroup>
</DrawingGroup>
```

**Comment by paulushub** [comment]
@LucaPassarella The transform issue is from the **Ensure Viewbox Position** option. If see to false, there is no problem, trying to find a way to fix this.

**Comment by paulushub** [comment]
@LucaPassarella I have provided a work around this issue. The translation to reposition the drawing should not resize the drawing. However, in this case it does. So, I check the effect of this translation and reset if it affects the drawing size.
Please give it a test.

**Comment by LucaPassarella** [comment]
Ok thanks very match. I'm testing it.

**Comment by LucaPassarella** [comment]
I found another case with the same problem, and the fix does not work.
Probably it's due to the position of the red circle.

[file.zip](https://github.com/ElinamLLC/SharpVectors/files/5506159/file.zip)

**Comment by paulushub** [comment]
@LucaPassarella The actual problem with these files is that they are way off the viewbox. The first file is off, but this is far too off the viewbox. Open it in Inkscape, if it does not show in the viewbox, then really it should not be shown by the SharpVectors.
The translation tries to make up for those off the viewbox drawings, but this one will not even show in browsers.

**Comment by LucaPassarella** [comment]
ok it's clear, thanks

**Comment by LucaPassarella** [comment]
I don't know if I was clear, have you try to change the stroke-width value. With 0.001 size the draw go out of the view, but with a bigger value it works.

**Comment by paulushub** [comment]
@LucaPassarella Yes, it was clear. On investigating, I found it was resulting from the translation transform applied after the drawing is rendered to try and position it within the viewbox.
As I stated, it was a work around the issue, since I could not tell why applying the transform changes the size in such case. What I did was to get the size of the drawing before and after applying the transform, if it changes, I discard the transform all together.
